File: Problem/codility/codility_Lesson03_Q3.cpp
#include <iostream>
#include <algorithm>
#include <cstdio>
#include <vector>
#include <string>

using namespace std;

int solution(int X, int Y, int D) {
int res;

res = (abs(Y - X) % D == 0 ? abs(Y - X) / D : abs(Y - X) / D + 1);

return res;
}

int main() {
int res = solution(10, 85, 30);

return 0;
}

File: Problem/codility/codility_Lesson03_Q3.java
class Solution {
    public int solution(int[] A) {

        int minVal = 9999999;
        int sum = 0;

        for(int i=0;i<A.length;i++){
            sum += A[i];
        }

        int left = 0;
        int right = sum;
        for(int i=1;i<A.length;i++){
            left += A[i-1];
            right -= A[i-1];

            minVal = Math.min(minVal, Math.abs(left-right));
        }

        return minVal;
    }
}